core/arch/arm/plat-k3/drivers/sa2ul_rng.c

```diff
@@ -125,22 +151,11 @@ TEE_Result hw_get_random_bytes(void *buf, size_t len)
 
 TEE_Result sa2ul_rng_init(void)
 {
-	uint32_t val = 0;
-
 	rng = (vaddr_t)phys_to_virt(RNG_BASE, MEM_AREA_IO_SEC, RNG_REG_SIZE);
 
-	/* Ensure initial latency */
-	val |= RNG_CONFIG_MIN_REFIL_CYCLES << RNG_CONFIG_MIN_REFIL_CYCLES_SHIFT;
-	val |= RNG_CONFIG_MAX_REFIL_CYCLES << RNG_CONFIG_MAX_REFIL_CYCLES_SHIFT;
-	io_write32(rng + RNG_CONFIG, val);
-
-	/* Configure the desired FROs */
-	io_write32(rng + RNG_FRODETUNE, 0x0);
-
-	/* Enable all FROs */
-	io_write32(rng + RNG_FROENABLE, 0xffffff);
-
-	io_write32(rng + RNG_CONTROL, ENABLE_TRNG);
+	/* Check if rng has already been initialized by a firmware */
```

Comment:
It might not be firmware, do we really trust the TRNG is setup correctly by just checking that it is enabled?

Previously we setup the firewall to protect our access to the TRNG and then unconditionally run setup on it. Now if someone (could be anyone, before we put up the firewall anyone in the system can mess with it (like R5 SPL)) set it up incorrectly (to return all 0s for instance) then we just trust it now..

Reply:
You're right.
It's more secure if we setup it unconditionally once the firewall is up.

Reply:
I updated the PR.
Now the TRNG is setup unconditionally during the driver init.